src/OpenConext/Profile/Value/Arp.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/**
+ * Copyright 2017 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\Profile\Value;
+
+use OpenConext\Profile\Exception\InvalidArpDataException;
+use Surfnet\SamlBundle\Exception\UnknownUrnException;
+use Surfnet\SamlBundle\SAML2\Attribute\Attribute;
+use Surfnet\SamlBundle\SAML2\Attribute\AttributeDefinition;
+use Surfnet\SamlBundle\SAML2\Attribute\AttributeDictionary;
+
+/**
+ * The Arp value object represents the Arp configuration for a given entity.
+ */
+final class Arp
+{
+    /**
+     * @var array The arp configuration is grouped on source. The source values are a collection of Attribute
+     */
+    private $arp;
+
+    public static function createWith(array $arp, AttributeDictionary $dictionary = null)
+    {
+        // Input validation
+        foreach ($arp as $attributeInformation) {
+            if (!is_array($attributeInformation)) {
+                throw new InvalidArpDataException('The attribute information in the arp should be an array.');
+            }
+            if (!self::isValidAttribute($attributeInformation)) {
+                throw new InvalidArpDataException('The attribute information is formatted invalidly.');
+            }
+        }
+
+        return new self($arp, $dictionary);
+    }
+
+    private function __construct(array $arp, AttributeDictionary $dictionary = null)
+    {
+        $arpCollection = [];
+
+        // Create Attribute instances for all attributes
+        foreach ($arp as $attributeName => $attributeDefinitionInformation) {
```

Comment:
Can't we lazy-load this?

Reply:
Can be done but is quite impractical to do so at this moment.